### PR TITLE
Jobs not services

### PR DESCRIPTION
**Pull request by tgross**
Without changing the configuration syntax in any way, this renames the package `services` to `jobs`, along with all related types. This will reflect the configuration changes in https://github.com/joyent/rfd/pull/31

### CODE REVIEW

File: jobs/config.go [hunk 1]
```diff
@@ -111,7 +111,7 @@ func NewConfigs(raw []interface{}, disc discovery.Backend) ([]*Config, error) {
 // Validate ensures that a Config meets all constraints
 func (cfg *Config) Validate(disc discovery.Backend) error {
 	if disc != nil {
-		// non-advertised services don't need to have their names validated
+		// non-advertised jobs don't need to have their names validated
 		if err := utils.ValidateServiceName(cfg.Name); err != nil {
```
Comment (geek) [inline jobs/config.go]: `ValidateServiceName` -> `ValidateJobName` ?

File: jobs/jobs.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+	restartLimit   int
+	restartsRemain int
+	frequency      time.Duration
+
+	events.EventHandler // Event handling
+}
+
+// NewJob creates a new Job from a Config
+func NewJob(cfg *Config) *Job {
+	service := &Job{
```
Comment (geek) [inline jobs/jobs.go]: `service` -> `job`

File: jobs/jobs.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+func FromConfigs(cfgs []*Config) []*Job {
+	services := []*Job{}
+	for _, cfg := range cfgs {
+		service := NewJob(cfg)
+		services = append(services, service)
+	}
+	return services
+}
+
+// SendHeartbeat sends a heartbeat for this service
```
Comment (geek) [inline jobs/jobs.go]: `// SendHeartbeat sends a heartbeat for this service` -> `// SendHeartbeat sends a heartbeat for this job` ?

File: jobs/config.go (hunk 1 above)
Comment (tgross) [inline jobs/config.go]: Not in that case, actually. We want to validate a service name in the Consul sense of the term.

File: jobs/jobs.go [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+func (job *Job) SendHeartbeat() {
+	if job.discoveryService != nil || job.Definition != nil {
```
Comment (geek) [inline jobs/jobs.go]: should we rename to `discoveryCatalog` ?

File: jobs/jobs.go (hunk 3 above)
Comment (tgross) [inline jobs/jobs.go]: Arguably not here either, as this is sends for service definition and not the job.

File: jobs/jobs.go (hunk 4 above)
Comment (tgross) [inline jobs/jobs.go]: Couldn't hurt

**Comment by tgross** [comment]
I think I've hit all your comments @geek and a couple others I'd missed. The big thing here to do was the package and struct rename so that my later changesets are smaller. 😀

**Comment by geek** [comment]
LGTM

**Pull request merged**